rtgui/filecatalog.cc

```diff
@@ -1704,17 +1707,15 @@ void FileCatalog::reparseDirectory ()
     }
 
     // check if a new file has been added
-    for (size_t i = 0; i < nfileNameList.size(); i++) {
-        bool found = false;
-
-        for (size_t j = 0; j < fileNameList.size(); j++)
-            if (nfileNameList[i] == fileNameList[j]) {
-                found = true;
-                break;
-            }
+    // build a set of collate-keys for faster search
+    std::set<std::string> oldNames;
```

Comment:
Have you considered `std::unordered_set<>`? Would be interesting, if it's faster for your workload...

Reply:
I just tested using an unordered_set. No difference in processing time. I guess the most expensive part here still are the collate_key() calls.